sycl/plugins/unified_runtime/pi_unified_runtime.cpp

```diff
@@ -1349,6 +1349,7 @@ __SYCL_EXPORT pi_result piPluginInit(pi_plugin *PluginInit) {
   }
 
   HANDLE_ERRORS(urLoaderInit(0, LoaderConfig));
+  HANDLE_ERRORS(urLoaderConfigRelease(LoaderConfig));

```

Comment:
This is an improvement, but if `urLoaderConfigEnableLayer` or `urLoaderInit` fail we still leak. Do we need to fix this RAII-style instead?

Reply:
I did recently add RAII wrappers to the UR CTS, there's deffo a case for making those first class citizens so they can be reused where needed.

Reply:
I will look into the RAII wrappers. Hi @kbenzie , do you have some links to some documents/commit that I can start with? Thanks.

Reply:
Hi @kbenzie , did UR alrdeay provide such RAII warppers or did you mean that you plan to add such wrappers to UR?

Reply:
We do not currently provide public RAII wrappers, I've created https://github.com/oneapi-src/unified-runtime/issues/1334 to address this.

I don't think this should block this PR.